newIDE/app/src/Export/Builds/BuildProgressAndActions.js

```diff
@@ -156,32 +154,53 @@ export default ({
             </Line>
           </React.Fragment>
         ) : build.status === 'pending' ? (
-          <Line alignItems="center" expand>
-            <LinearProgress
-              style={{ flex: 1 }}
-              value={
-                config.estimatedTimeInSeconds(build) > 0
-                  ? ((config.estimatedTimeInSeconds(build) -
-                      estimatedRemainingTime) /
-                      config.estimatedTimeInSeconds(build)) *
-                    100
-                  : 0
-              }
-              variant={
-                estimatedRemainingTime > 0 ? 'determinate' : 'indeterminate'
-              }
-            />
-            <Spacer />
-            {estimatedRemainingTime > 0 ? (
+          <Line alignItems="center" expand justifyContent="center">
+            {estimatedRemainingTime > -estimatedTime && (
+              <>
+                <LinearProgress
+                  style={{ flex: 1 }}
+                  value={
+                    estimatedTime > 0
+                      ? ((estimatedTime - estimatedRemainingTime) /
+                          estimatedTime) *
+                        100
+                      : 0
+                  }
+                  variant={
+                    estimatedRemainingTime > 0 ? 'determinate' : 'indeterminate'
+                  }
+                />
+                <Spacer />
+              </>
+            )}
+            {estimatedRemainingTime > 0 && (
```

Comment:
I'm having troubles figuring out if one of those 4 conditioned rendering could be rendered at the same time.
Maybe can you name the booleans `const isLongerThanEstimatedButNotTooMuch = estimatedRemainingTime < 0 &&
              estimatedRemainingTime > -estimatedTime` or something simpler 😅
Or have a `if` with `else`s to clearly what rendering can happen together or not.

Reply:
yep, I like that 👍  will do